src/test/munit/directory-listener-functional-test-case.xml

```
@@ -0,0 +1,396 @@
+        <file:create-directory config-ref="file-list-config" directoryPath="withMatcher"/>
+    </munit:before-test>
+
+    <sub-flow name="storeMessages">
+        <set-variable value="#[message]" variableName="message"/>
+        <munit-tools:retrieve  key="messages"/>
+        <munit-tools:store key="messages">
+            <munit-tools:value >#[
+                %dw 2.0
+                output application/java
+                ---
+                payload ++ [vars.message]
+                ]
+            </munit-tools:value>
+        </munit-tools:store>
+    </sub-flow>
+
+    <flow name="source-on-file-created-test" >
+        <file:listener config-ref="file-list-config" directory="matcherless" recursive="false">
+            <scheduling-strategy >
+                <fixed-frequency />
+            </scheduling-strategy>
+        </file:listener>
+        <flow-ref name="storeMessages"/>
+    </flow>
+    <munit:test name="on-file-created-test" description="Verifies that a created file is picked">
+        <munit:enable-flow-sources >
+            <munit:enable-flow-source value="source-on-file-created-test"/>
+        </munit:enable-flow-sources>
+        <munit:behavior >
+            <file:write config-ref="file-list-config" path="matcherless/watchme.txt">
+                <file:content>who watches the watchmen?</file:content>
+            </file:write>
+        </munit:behavior>
+        <munit:execution>
+            <try>
+                <until-successful maxRetries="10" millisBetweenRetries="1000">
+                    <munit-tools:retrieve key="messages"/>
+                    <validation:validate-size value="#[payload]" min="1"/>
+                </until-successful>
+                <error-handler >
+                    <on-error-continue enableNotifications="false" logException="false" type="ANY">
+                        <munit-tools:retrieve key="messages"/>
```

Why do we keep running the test if the until-successful fails? The same happens on other tests

---

Do you mean why `<munit-tools:retrieve key="messages"/>` is after `on-error-continue`?

---

Yes, why not just let the test fail?

---

I just copied the structure from another test and forgot to remove that.